keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/annealing/ResourcePersister.kt

```diff
@@ -1,44 +1,52 @@
 package com.netflix.spinnaker.keel.annealing
 
 import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceName
+import com.netflix.spinnaker.keel.api.SubmittedResource
 import com.netflix.spinnaker.keel.events.ResourceCreated
-import com.netflix.spinnaker.keel.events.ResourceDeleted
-import com.netflix.spinnaker.keel.events.ResourceEvent
 import com.netflix.spinnaker.keel.events.ResourceUpdated
 import com.netflix.spinnaker.keel.persistence.ResourceRepository
 import com.netflix.spinnaker.keel.persistence.get
 import com.netflix.spinnaker.keel.plugin.ResourceHandler
 import com.netflix.spinnaker.keel.plugin.supporting
 import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationEventPublisher
 import org.springframework.stereotype.Component
+import java.time.Clock
 
 @Component
 class ResourcePersister(
   private val resourceRepository: ResourceRepository,
   private val handlers: List<ResourceHandler<*>>,
-  private val queue: ResourceCheckQueue
+  private val queue: ResourceCheckQueue,
+  private val publisher: ApplicationEventPublisher,
+  private val clock: Clock
 ) {
-  fun handle(event: ResourceEvent): Resource<*> {
-    log.info("Received event {}", event)
-    return when (event) {
-      is ResourceCreated ->
-        handlers.supporting(event.resource.apiVersion, event.resource.kind)
-          .normalize(event.resource)
-          .also(resourceRepository::store)
-          .also { queue.scheduleCheck(it) }
-      is ResourceUpdated ->
-        handlers.supporting(event.resource.apiVersion, event.resource.kind)
-          .normalize(event.resource)
-          .also(resourceRepository::store)
-          .also { queue.scheduleCheck(it) }
-      is ResourceDeleted -> {
-        resourceRepository.delete(event.name)
-        resourceRepository.get<Any>(event.name).also {
-          queue.scheduleCheck(it)
-        }
-      }
+  fun create(resource: SubmittedResource<Any>): Resource<out Any> =
+    handlers.supporting(resource.apiVersion, resource.kind)
+      .normalize(resource)
+      .also(resourceRepository::store)
+      .also { resourceRepository.appendHistory(ResourceCreated(it, clock)) }
+      .also { queue.scheduleCheck(it) }
+
+  fun update(resource: Resource<Any>): Resource<out Any> {
+    val existing = resourceRepository.get<Any>(resource.metadata.uid)
+
+    return if (existing.spec == resource.spec) {
```

Comment:
Eventually this will become where we diff the resources & store the delta in the event. For now straight equality check will do.